### ISSUE TITLE
 ignoring return value in test

### ISSUE THREAD

**Issue by t-hishinuma** [ricosjp/monolish#74]
```
matrix_transpose.cpp:60:3: warning: ignoring return value of 'monolish::matrix::COO<Float>& monolish::matrix::COO<Float>::transpose() [with Float = double]', declared with attribute nodiscard [-Wunused-result]
   60 |   A.transpose();
```

**Comment by t-hishinuma** [comment]
```
matrix_common.cpp:79:5: warning: ignoring return value of 'monolish::matrix::COO<Float>& monolish::matrix::COO<Float>::transpose() [with Float = double]', declared with attribute nodiscard [-Wunused-result]
   79 |     transposed_COO1.transpose();

```

**Comment by t-hishinuma** [comment]
The implementation of transpose(), which is the cause of the warning, is wrong, so it will be fixed in #77.
